packages/@lwc/engine/src/framework/main.ts

```diff
@@ -28,3 +28,5 @@ export { readonly } from './readonly';
 export { deprecatedBuildCustomElementConstructor as buildCustomElementConstructor } from './wc';
 
 export { setFeatureFlag, setFeatureFlagForTest } from '@lwc/features';
+
+export { registerTemplateSwap } from './hot-swaps';
```

Comment:
@diervo we need to restrict this in platform, and this is the one we call with the old template and the new template functions from outside.